### DISCUSSION TITLE
Measuring times (like convergence time) by Contiki-ng/Cooja

### DISCUSSION THREAD

**Discussion by h-hosseini** [contiki-ng/contiki-ng#2927] [General]
We (Dr. @ElisaRojas and @davidcawork) want to measure convergence time. Since nodes require a global timer (and variable maybe) to measure this metric, and implementing such an approach will waste the node's resources, we went to Cooja to reach this goal. Our approach to measure the metric was to parse log files and use time timestamps. Since the IO delay may affects on the logs (i.e. IO devices are slow and lazy, and logs or IO commands are usually implemented by IO interrupts which may have less priority than other interrupts ...), there are some questions for us!
Are there such delays in the logs of Cooja?
Is there a more accurate approach to measure such a metric in Contiki-ng and Cooja?

**Comment by yanpads** [comment]
The time between `Cooja log` and `radio message log` are slightly the same. If you just select one between the two and stick to it then there will be no problem.

**Comment by h-hosseini** [comment]
Dear @yanpads ,

Thanks for your response!

About the "slightly" term you mentioned already, If you explain a bit about why "slightly" arises, I will be grateful. Is this difference because of the nature of interrupt or is there any other reason?

**Comment by yanpads** [comment]
You can check the time of the `Cooja log` and `radio message log` for every receive or transmit packet. They are different.
